qiskit/dagcircuit/dagcircuit.py

```diff
@@ -1393,6 +1393,28 @@ def substitute_node(self, node, op, inplace=False):
             self._decrement_op(node.op)
         return new_node
 
+    def weakly_connected_components(self) -> List["DAGCircuit"]:
+        """Decompose the dag circuit into its weakly connected components."""
+        connected_components = rx.weakly_connected_components(self._multi_graph)
+
+        # Collect each disconnected subgraph
+        disconnected_subgraphs = []
+        for components in connected_components:
+            disconnected_subgraphs.append(self._multi_graph.subgraph(list(components)))
+
+        # Create new DAGCircuit objects from each of the rustworkx subgraph objects
+        decomposed_dags = []
+        for subgraph in disconnected_subgraphs:
+            new_dag = DAGCircuit()
+            new_dag.add_qubits(self.qubits)
```

Comment:
Two things here:

1. this erases register information, and ignores any classical bits.  The graph partition will account for clbits, and may well even produce some graphs that _only_ have clbits, which likely isn't what you want.  I'm guessing from your perspective, you might not care about clbits being shared between the different components if you're running the separate circuits on different parts of the chip, because the classical links are free(er - networking the control electronics across the chips is still pretty non-trivial).
2. I could see arguments in both directions about the output DAGs being full-width in qubits or only containing the relevant bits.  Perhaps this is something we'll want a keyword argument to decide?

Reply:
After talking with @kdk , I've decided to leave all qubits and clbits in place in the output sub-DAGs using `copy_empty_like`, so all sub-DAGs will have same number of qubits/clbits. However, I have removed any sub-DAGs generated from empty qubits or clbits.

Reply:
I didn't realize there was no QuantumCircuit method for removing idle wires. I'm not sure we should leave idle wires if users will just have to return to the DAG to remove them again.

I think we should clean up idle wires in the sub-DAGs. Thoughts?

Reply:
My two cents is that this feels like two separate steps. The first separates the `DAGCircuit` into a series of similarly shaped dags, and then if the user wants, they can subsequently e.g. remove idle wires/reindex qubits to be zero based/remove register information (possibly before they convert back to a `QuantumCircuit`).

Also, there isn't a reason I know of that these functions couldn't also exist on the `QuantumCircuit` (either directly or as passthroughs to their DAG implementations), but agree it's unfortunate they don't exist now.

Reply:
The structure of `DAGCircuit` means that "find the idle wires" is linear complexity in the number of wires, but in `QuantumCircuit` you've got to iterate over every instruction.  It's simple logic on both, but the data structure makes it natural as an "easy" operation on `DAGCircuit`, which is probably why it's here but not on `QuantumCircuit`.

Reply:
I agree that having a `separable_circuits` and `clean_idle_wires` would be useful methods `QuantumCircuit`, even if you get hit with the overhead of converting to and from DAG.

I could implement those two methods in `QuantumCircuit` as well, if you agree they would be useful.

Reply:
On `QuantumCircuit` we use the terminology "bits" not "wires", but otherwise that sounds fine to me, if you and Kevin think it's useful.

Reply:
I almost feel like `clean_idle_bits` should be a flag for `separable_circuits`, so that you can do both while you have the DAG instead of getting hit with two copies. We can discuss more in the PR, if we decide to do it, but.. :)